### ISSUE TITLE
Xdebug not properly working on ubuntu-12.04 (dev)

### ISSUE THREAD

**Issue by simonschaufi** [webdevops/Dockerfile#110]
Hi, @julianxhokaxhiu was setting up the docker setup for the company I work for. This includes a  ``.user.ini`` file and a ssh tunnel.

When I use debian-8 (PHP Version 5.6.22-0+deb8u1), xdebug works.
When I use ubuntu-12.04 (PHP Version 5.3.10-1ubuntu3.23), xdebug is not working.

Both versions come from the -dev branch.

I use the following ``.user.ini`` file in my project's root directory:

```ini
xdebug.remote_host=172.17.0.1
xdebug.remote_connect_back=0
xdebug.remote_autostart=1
```
Even if I add ``xdebug.idekey=docker``, it doesn't take effect.

Here are the diffs (from phpinfo()):

| Setting | Ubuntu | Debian |
|----------------------------|-------------|--------|
| xdebug.idekey | application | docker |
| xdebug.remote_connect_back | On | Off |

**Comment by mblaschke** [comment]
Which version of docker images do you use?
Can you show us `docker images |grep webdevops`?

**Comment by simonschaufi** [comment]
```
webdevops/php-apache-dev:debian-8
webdevops/php-apache-dev:ubuntu-12.04
```

**Comment by mblaschke** [comment]
Please post the whole line with the age/created date

**Comment by simonschaufi** [comment]
updated

**Comment by mblaschke** [comment]
will check it later

**Comment by simonschaufi** [comment]
I checked one more thing:
with ubuntu-15.10 it works. So I think it is just this exact version that is not working.

**Comment by julianxhokaxhiu** [comment]
It could may be that PHP 5.3 xdebug version does not accept connect back maybe?

**Comment by mblaschke** [comment]
Does it work when you upload your php.ini to `/opt/docker/etc/php/php.ini`?

PS: PHP 5.3 is EOL since mid 2014

**Comment by simonschaufi** [comment]
Problem solved!
The idekey is not needed, so I just added the line
```ini
xdebug.remote_connect_back=Off
```
into the file ``/opt/docker/etc/php/php.ini``, restarted docker and then it worked!

PS: is there a better way to restart apache so that the changes in the php.ini file take effect? I tried it with:
```bash
/etc/init.d/apache2 restart
```
this was the output:
```bash
 * Restarting web server apache2
 ... waiting httpd (pid 1804) already running              [ OK ]
```
but the changes from the ini file didn't take effect.

**Comment by julianxhokaxhiu** [comment]
Can you please try

```
xdebug.remote_host=172.17.0.1 xdebug.remote_connect_back=Off xdebug.remote_autostart=On
```

in your `.user.ini`?

On a clean docker please. So destroy and recreate it. Thanks!

**Comment by simonschaufi** [comment]
No, not working.

this is the output of phpinfo (which is wrong):
```ini
xdebug.remote_connect_back	On
```

**Comment by julianxhokaxhiu** [comment]
Right guess! i think this is the real bug :) or maybe missing feature in 5.3? But I don't think so...

**Comment by mblaschke** [comment]
I've created a PR with environment variables for xdebug and blackfire
https://github.com/webdevops/Dockerfile/pull/115

**Comment by julianxhokaxhiu** [comment]
Environments are another solution which is kinda cool, but It would be interesting to inspect `.user.ini` files since they could be different per project.

Do you think is possible to do so?

**Comment by guiyomh** [comment]
You can run a another container with a different conf.

**Comment by julianxhokaxhiu** [comment]
That's also true, but why not inspecting the `.user.ini` possibility? :P

**Comment by mblaschke** [comment]
hmm... I don't have any idea why it's not working for ubuntu 12.04

**Comment by simonschaufi** [comment]
Where is the file loaded actually?

**Comment by guiyomh** [comment]
@julianxhokaxhiu : I think that docker's philosophy, is to wrap an application and its requirements in one containers, to ease its deployment. If your application has different requirements. Then for me, she has to be in another container. (IHMO)

**Comment by guiyomh** [comment]
PS : I have accepted the PR

**Comment by julianxhokaxhiu** [comment]
You're totally right, although I'm just suggesting another method that could be useful to be investigated :) anyway it's up to you to decide

**Comment by simonschaufi** [comment]
To make use of this change, do I need to destroy my docker and create a new one from the updated image?

**Comment by mblaschke** [comment]
These images are not built yet so they are not available on docker hub.
Will try to make a new release this weekend to it will be available on monday.

You have to pull the image with `docker pull [imagename]` and rebuild your container with it.

**Comment by simonschaufi** [comment]
Good morning, I see an update of my docker image but the expected additional lines in ``/opt/docker/etc/php/php.ini`` are still not there.

**Comment by simonschaufi** [comment]
this is the script that Julian wrote:

```
sudo docker run \
    -d \
    --restart=always \
    --name="$FULLDOMAIN" \
    --link="mysql-db:db" \
    -e "VIRTUAL_HOST=$FULLDOMAIN" \
    -v "/home/dev/docker/$FULLDOMAIN:/app" \
    "webdevops/php-apache-dev:$VERSION" &>/dev/null
```
where FULLDOMAIN looks like project.developer.lan

What do I need to change?

**Comment by simonschaufi** [comment]
solved with #115
